compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/PersistRecordValidator.java

```diff
@@ -180,12 +180,24 @@ private Optional<AnnotationNode> getEntityAnnotation(NodeList<AnnotationNode> an
         return Optional.empty();
     }
 
-    private boolean isDuplicateEntity(Entity entity, TypeDefinitionNode typeDefinitionNode) {
-        if (this.entities.containsKey(entity.getEntityName())) {
-            String initialModule = this.entities.get(entity.getEntityName()).getModule();
-            entity.addDiagnostic(typeDefinitionNode.typeName().location(), DiagnosticsCodes.PERSIST_119.getCode(),
-                    MessageFormat.format(DiagnosticsCodes.PERSIST_119.getMessage(), initialModule),
-                    DiagnosticsCodes.PERSIST_119.getSeverity());
+    private void validateEntityInMultipleSubModule(Entity entity) {
+        if (this.isEntityInMultipleModules) {
+            entity.addDiagnostic(entity.getLocation(), DiagnosticsCodes.PERSIST_119.getCode(),
+                    DiagnosticsCodes.PERSIST_119.getMessage(), DiagnosticsCodes.PERSIST_119.getSeverity());
+        } else if (entityInMultipleSubModules(entity)) {
+            for (Map.Entry<String, Entity> entry : this.entities.entrySet()) {
+                entity.addDiagnostic(entry.getValue().getLocation(), DiagnosticsCodes.PERSIST_119.getCode(),
+                        DiagnosticsCodes.PERSIST_119.getMessage(), DiagnosticsCodes.PERSIST_119.getSeverity());
+            }
+            entity.addDiagnostic(entity.getLocation(), DiagnosticsCodes.PERSIST_119.getCode(),
+                    DiagnosticsCodes.PERSIST_119.getMessage(), DiagnosticsCodes.PERSIST_119.getSeverity());
+        }
+    }
+
+    private boolean entityInMultipleSubModules(Entity entity) {
```

Comment:
Is separate function necessary, I feel this can be an inline check

Reply:
Updated in 45dc85e